### PR TITLE
Adding Automated Connection Tracking Test Cases

### PR DESCRIPTION
**Pull request by hirentha**
Signed-off-by: Hiren Thakkar <hiren.thakkar@intel.com>

### CODE REVIEW

**Comment by hirentha** [comment]
Attached All Scripts Success Logs.
[CT_Test_Cases_Logs.txt](https://github.com/ipdk-io/ovs/files/9434144/CT_Test_Cases_Logs.txt)

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-1/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

File: Makefile.am [hunk 1]
```diff
@@ -215,6 +221,11 @@ EXTRA_DIST = \
 	ptf_tests/common/p4c_artifacts/linux_networking/tunnel.p4 \
 	ptf_tests/common/p4c_artifacts/l3_exact_match_add_del/l3_exact_match_add_del.conf \
 	ptf_tests/common/p4c_artifacts/l3_exact_match_add_del/l3_exact_match_add_del.p4 \
+        ptf_tests/common/p4c_artifacts/pna_tcp_connection_tracking/network_port_config.json \
+        ptf_tests/common/p4c_artifacts/pna_tcp_connection_tracking/pipe/.gitkeep \
```
Comment (n-sandeep) [inline Makefile.am]: Looks like this file is not present.
Build is failing because of this.

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-2/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-3/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-4/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-5/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-6/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-7/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-8/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-9/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/e3912fbb7759f8158575b02855e5944307347b3b

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-10/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/1aa1cac12c3ef6eb92416072390baa000b36fd8d

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-11/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/94ad571acf9a42b070e7c7265550168b2dda7b5d

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-12/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/191dd962ed2edca8d17c115c014a47204cf0168b

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-13/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/191dd962ed2edca8d17c115c014a47204cf0168b

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-14/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/5a7255dab8729dfbcb845e7e7a8e98a8049698da

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-15/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/5a7255dab8729dfbcb845e7e7a8e98a8049698da

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-16/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/5a7255dab8729dfbcb845e7e7a8e98a8049698da

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-17/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/5a7255dab8729dfbcb845e7e7a8e98a8049698da

File: ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+"""
+DPDK Connection Tracking with Link Port
```
Comment (dkaranjai) [inline ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py]: correct the name

File: ptf_tests/tests/ct_rule_del_add_test.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
+"""
+DPDK Connection Tracking with Link Port
```
Comment (dkaranjai) [inline ptf_tests/tests/ct_rule_del_add_test.py]: correct the name

File: ptf_tests/tests/ct_single_connection_tap_link_ports.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
+"""
+DPDK Connection Tracking with Link Port
```
Comment (dkaranjai) [inline ptf_tests/tests/ct_single_connection_tap_link_ports.py]: correct the name

**Review by dkaranjai — CHANGES_REQUESTED** [review]
please attach the p4ovs build log.

**Comment by hirentha** [comment]
Attached ipdk building logs.
[PR_59_Build_Logs.txt](https://github.com/ipdk-io/ovs/files/9493214/PR_59_Build_Logs.txt)

File: ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py (hunk 2 above)
Comment (hirentha) [reply ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py]: Updated.

File: ptf_tests/tests/ct_single_connection_tap_link_ports.py (hunk 4 above)
Comment (hirentha) [reply ptf_tests/tests/ct_single_connection_tap_link_ports.py]: updated

File: ptf_tests/tests/ct_rule_del_add_test.py (hunk 3 above)
Comment (hirentha) [reply ptf_tests/tests/ct_rule_del_add_test.py]: updated

**Review by dkaranjai — APPROVED** [review]
LGTM

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-18/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2c3990d07b362286bd082d477a1ea412e20ec266

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-19/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2c3990d07b362286bd082d477a1ea412e20ec266

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-20/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2c3990d07b362286bd082d477a1ea412e20ec266

File: ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            print(match_action)
+            if not ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                   print("-----------------------------------------------------")
+                   print("Above Fail is  as expected for Negative Test Exection")
+                   print("-----------------------------------------------------")
+                   
+                                      
+
+        # Sleep 5 Sec before tcp connection establishment
+        time.sleep(5)
```
Comment (sijubaker) [inline ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py]: not required. as no tcp handshake happening

File: ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    "traffic": 
+  {
+          "type": "tcp",
+          "send_port": [0, 1],
+          "receive_port": [1, 0],
+          "in_pkt_header": {
+    "eth_mac": ["00:00:01:00:00:01","00:00:02:00:00:01"],
+    "ip_address" : ["10.10.10.10","20.20.20.10"],
+    "tcp_port" : [10001, 20001],
+    "flags" : ["SA", "A", "F", "FA", "R"]
```
Comment (sijubaker) [inline ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json]: flags not used in script. can be removed..

File: ptf_tests/tests/ct_rule_del_add_test.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+ 
+        table = self.config_data['table'][1]
+        print(f"Rule Creation : {table['description']}")
+        print(f"Adding {table['description']} rules")
+        for match_action in table['match_action']:
+            if not ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
+        
+        # Sleep 5 Sec before tcp connection establishment
```
Comment (sijubaker) [inline ptf_tests/tests/ct_rule_del_add_test.py:103]: sleep for tcp establishment not required here.

File: ptf_tests/tests/ct_rule_del_add_test.py [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
+ 
+        table = self.config_data['table'][1]
+        print(f"Rule Creation : {table['description']}")
+        print(f"Adding {table['description']} rules")
+        for match_action in table['match_action']:
+            if not ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
```
Comment (sijubaker) [inline ptf_tests/tests/ct_rule_del_add_test.py:101]: traffic check can be added after adding rules

File: ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json [hunk 9]
```diff
@@ -0,0 +1,82 @@
+{
```
Comment (sijubaker) [inline ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json]: please follow similar naming as other scripts "all lower case"

**Review by sijubaker — COMMENTED** [review]
General comments :

1. for multiple_connection test case have multiple connections existing at same time. Not sequential
2. No aging tests covered. Need to check for timeouts..

File: ptf_tests/common/utils/test_utils.py [hunk 10]
```diff
@@ -639,3 +664,187 @@ def ip_ntns_exec_ping_test(nsname, dst_ip, count="4", remote=False, hostname="",
...
+    output = conn.readResult().split("\n")[1]
+    
+    if "Netperf version" not in output:
+        print (f"FAILED: netperf is not installed on {vmname}")
+        return False
+    
+    print (f"PASS: verify netperf is installed on {vmname}")
+    return True
+
+def vm_check_netserver(conn):
```
Comment (sijubaker) [inline ptf_tests/common/utils/test_utils.py]: the util can be modified with more suitable name. This proc is mainly used to "start" netserver not just check the status.

File: ptf_tests/common/utils/test_utils.py [hunk 11]
```diff
@@ -639,3 +664,187 @@ def ip_ntns_exec_ping_test(nsname, dst_ip, count="4", remote=False, hostname="",
...
+        net_server_cmd ="netserver" 
+        conn.sendCmd(net_server_cmd)
+        output = conn.readResult().split("\n")
+        if len(output) < 4:
+            print(f"FAIL: Unable to start netserver")
+            return False
+    #print ("PASS: netserver is running on VM") 
+    return True
+
+def vm_netperf_client(conn, host, testlen, testname, option=""):
```
Comment (sijubaker) [inline ptf_tests/common/utils/test_utils.py]: output of this util is not very clear. Either format it or print only the values which user is interested into (eg:throughput)

**Review by sijubaker — CHANGES_REQUESTED** [review]
Please address the review comments given..

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-21/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2c3990d07b362286bd082d477a1ea412e20ec266

File: ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py (hunk 5 above)
Comment (hirentha) [reply ptf_tests/tests/Nt_Ct_unsupported_arguments_values.py]: updated

File: ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json (hunk 6 above)
Comment (hirentha) [reply ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json]: updated

File: ptf_tests/tests/ct_rule_del_add_test.py (hunk 7 above)
Comment (hirentha) [reply ptf_tests/tests/ct_rule_del_add_test.py:103]: updated

File: ptf_tests/tests/ct_rule_del_add_test.py (hunk 8 above)
Comment (hirentha) [reply ptf_tests/tests/ct_rule_del_add_test.py:101]: updated

File: ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json (hunk 9 above)
Comment (hirentha) [reply ptf_tests/common/config/Nt_Ct_unsupported_arguments_values.json]: updated

File: ptf_tests/common/utils/test_utils.py (hunk 10 above)
Comment (hirentha) [reply ptf_tests/common/utils/test_utils.py]: Needs to done by Shujun and after will apply in this PR.

File: ptf_tests/common/utils/test_utils.py (hunk 11 above)
Comment (hirentha) [reply ptf_tests/common/utils/test_utils.py]: Needs to done by Shujun and after will apply in this PR.

**Comment by hirentha** [comment]
Attached ovs building logs and Test cases run logs after made changes as per Siju's comment. 
[Build_logs_after_make_changes.txt](https://github.com/ipdk-io/ovs/files/9525499/Build_logs_after_make_changes.txt)
[ptf_ct_tests_after_required_changes.log](https://github.com/ipdk-io/ovs/files/9525501/ptf_ct_tests_after_required_changes.log)

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-22/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/c960c30b39ff1da7ab93cb67116dd00d5d48f102

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-23/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/1dedc3d940e00f514124e8cc50dcb3c4e93bd47a

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-24/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/cb2aad4ed62583dd0aeca9aad5b7610ef7938c85

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-25/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/cb2aad4ed62583dd0aeca9aad5b7610ef7938c85

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-26/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/4343a5042f78be13e3b2be7c2f8156fc25ba8936

**Comment by hirentha** [comment]
Hi Siju/ Dwaipayan,

I have completed all netperf related changes and added ct timer test case.
Attached building ovs logs with all changes and ran all scripts logs.
  
====================
Summary
====================
ct_single_connection_tap_link_ports : PASSED
ct_multiple_connection_tap_link_ports : PASSED
ct_rule_del_add_test : PASSED
nt_ct_unsupported_arguments_values : PASSED
ct_single_connection_vhost_ports_with_netperf : PASSED
ct_multiple_connection_vhost_ports_with_netperf : PASSED
ct_timer_test : PASSED


Ran 7 test(s), 7 Passed, 0 Failed.


[ptf_ct_tests.log](https://github.com/ipdk-io/ovs/files/9565285/ptf_ct_tests.log)
[Ct_7_test_cases_building_logs.txt](https://github.com/ipdk-io/ovs/files/9565298/Ct_7_test_cases_building_logs.txt)

File: ptf_tests/common/utils/test_utils.py [hunk 12]
```diff
@@ -37,144 +37,196 @@
 
 
 def add_port_to_dataplane(port_list):
-    """
```
Comment (dkaranjai) [inline ptf_tests/common/utils/test_utils.py]: please check this file. seems all required indentations are removed

File: set_hugepages.sh [hunk 13]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (dkaranjai) [inline set_hugepages.sh:17]: do we need this?

**Review by dkaranjai — CHANGES_REQUESTED** [review]

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-28/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ead9c470adcf669e1b7cc9b1b28c402dcb2316eb

File: ptf_tests/common/utils/test_utils.py (hunk 12 above)
Comment (hirentha) [reply ptf_tests/common/utils/test_utils.py]: updated

File: set_hugepages.sh (hunk 13 above)
Comment (hirentha) [reply set_hugepages.sh:17]: It's required as, it's not present in ipdk diectory.

**Review by dkaranjai — APPROVED** [review]

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-30/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/12ca04383ef203a6506f140e6d59ac0abdeb891c

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-31/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/12ca04383ef203a6506f140e6d59ac0abdeb891c

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-32/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ca9a072f394684f07076b782637a591a8c252c20

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-33/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ca9a072f394684f07076b782637a591a8c252c20

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-34/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/d8c77167902a521a58713ee338906861f33a2f4b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-35/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/6e6cb2b125bb49a6491e0e9c5bd91e8cc6dbd96b

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-36/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ed6e8a590271deb2fda3ac6b905ac0928d14e769

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-37/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ed6e8a590271deb2fda3ac6b905ac0928d14e769

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-38/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/ed6e8a590271deb2fda3ac6b905ac0928d14e769

**Review by sijubaker — APPROVED** [review]

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-39/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/c8097015a5b5ac7254ef3221078c067bdba2b246

**Comment by dkaranjai** [comment]
below line in Makefile.am is not complete

ideally it should end with an '\'



ptf_tests/common/p4c_artifacts/pna_tcp_connection_tracking/pna_tcp_connection_tracking.p

do we have this file?, if not we can remove this from Makefile.am

ptf_tests/common/p4c_artifacts/pna_tcp_connection_tracking/pipe/.gitkeep

Following seen in build log. 
[2022-10-03T09:25:19.250Z] make[2]: *** No rule to make target 'ptf_tests/common/p4c_artifacts/pna_tcp_connection_tracking/pipe/.gitkeep', needed by 'all-am'.  Stop.
[2022-10-03T09:25:19.250Z] make[2]: *** Waiting for unfinished jobs....

File: set_hugepages.sh [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+#Copyright (C) 2021 Intel Corporation
```
Comment (dkaranjai) [inline set_hugepages.sh]: this copyright info is wrong. should be 2022

**Review by dkaranjai — CHANGES_REQUESTED** [review]

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-40/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/c8097015a5b5ac7254ef3221078c067bdba2b246

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-41/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/336df5d52a6f3115d70dff7968edbbe657e17425

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-42/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/336df5d52a6f3115d70dff7968edbbe657e17425

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-43/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/01f5e6ab64c936212e4b226de508bbf1b1490421

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-44/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/01f5e6ab64c936212e4b226de508bbf1b1490421

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-45/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/01f5e6ab64c936212e4b226de508bbf1b1490421

**Review by dkaranjai — APPROVED** [review]

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-46/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/12f3c577c2beec74886db707c416085c16aa73a6

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-47/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2a4dadcfa3f5d31df125e96db36dde33b10a6e86

**Comment by ipdksw** [comment]
Result:ABORTED JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-48/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2a4dadcfa3f5d31df125e96db36dde33b10a6e86

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-59-49/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/2a4dadcfa3f5d31df125e96db36dde33b10a6e86

**Pull request merged**